task/servicetest/servicetest.go

```diff
@@ -37,17 +37,109 @@ func init() {
 // If creating the System fails, the implementer should call t.Fatal.
 type BackendComponentFactory func(t *testing.T) (*System, context.CancelFunc)
 
+// UsePlatformAdaptor allows you to set the platform adaptor as your TaskService.
+func UsePlatformAdaptor(s backend.Store, lr backend.LogReader, rc task.RunController, i *inmem.Service) influxdb.TaskService {
+	return task.PlatformAdapter(s, lr, rc, i, i, i)
+}
+
+// TaskControlAdaptor creates a TaskControlService for the older TaskStore system.
+func TaskControlAdaptor(s backend.Store, lw backend.LogWriter, lr backend.LogReader) backend.TaskControlService {
+	return &taskControlAdaptor{s, lw, lr}
+}
+
+// taskControlAdaptor adapts a backend.Store and log readers and writers to implement the task control service.
+type taskControlAdaptor struct {
+	s  backend.Store
+	lw backend.LogWriter
+	lr backend.LogReader
+}
+
+func (tcs *taskControlAdaptor) CreateNextRun(ctx context.Context, taskID influxdb.ID, now int64) (backend.RunCreation, error) {
+	return tcs.s.CreateNextRun(ctx, taskID, now)
+}
+func (tcs *taskControlAdaptor) FinishRun(ctx context.Context, taskID, runID influxdb.ID) (*influxdb.Run, error) {
+	// the tests arent looking for a returned Run because the old system didn't return one
+	// Once we completely switch over to the new system we can look at the returned run in the tests.
+	return nil, tcs.s.FinishRun(ctx, taskID, runID)
+}
+func (tcs *taskControlAdaptor) NextDueRun(ctx context.Context, taskID influxdb.ID) (int64, error) {
+	_, m, err := tcs.s.FindTaskByIDWithMeta(ctx, taskID)
+	if err != nil {
+		return 0, err
+	}
+	return m.NextDueRun()
+}
+func (tcs *taskControlAdaptor) UpdateRunState(ctx context.Context, taskID, runID influxdb.ID, when time.Time, state backend.RunStatus) error {
+	st, m, err := tcs.s.FindTaskByIDWithMeta(ctx, taskID)
+	if err != nil {
+		return err
+	}
+	var (
+		schedFor, reqAt time.Time
+	)
+	// check the log store
+	r, err := tcs.lr.FindRunByID(ctx, st.Org, runID)
+	if err == nil {
+		schedFor, _ = time.Parse(time.RFC3339, r.ScheduledFor)
+		reqAt, _ = time.Parse(time.RFC3339, r.RequestedAt)
+	}
+
+	// in the old system the log store may not have the run until after the first
+	// state update, so we will need to pull the currently running.
+	tt := time.Time{}
+	if schedFor == tt {
```

Comment:
Prefer `if schedFor.IsZero()`.

Reply:
Oo yah I knew that was somewhere. Thanks.